### PR TITLE
Localize git infrastructure.

### PR DESCRIPTION
**Pull request by BillyONeal**

### CODE REVIEW

**Comment by BillyONeal** [comment]
OSX hopefully fixed by https://github.com/microsoft/vcpkg-tool/pull/915

File: src/vcpkg/vcpkgpaths.cpp [hunk 1]
```diff
@@ -1011,38 +1013,41 @@ namespace vcpkg
         auto maybe_tar_output = flatten(cmd_execute_and_capture_output(tar_cmd_builder), Tools::TAR);
         if (!maybe_tar_output)
         {
-            auto message =
-                Strings::concat(PRELUDE, "Error: Failed to tar port directory\n", std::move(maybe_tar_output).error());
-
+            auto message = msg::format_error(msgGitCommandFailed, msg::command_line = tar_cmd_builder.command_line());
```
Comment (ras0219-msft) [inline src/vcpkg/vcpkgpaths.cpp:1016]: This no longer includes `maybe_tar_output` in the output.

File: src/vcpkg/vcpkgpaths.cpp [hunk 2]
```diff
@@ -1077,28 +1082,36 @@ namespace vcpkg
...
+                        .append_raw(line);
                 }
 
                 ret.emplace(split_line[1], file_info_section.back());
             }
             return ret;
         }
 
-        return Strings::format("Error: Couldn't get local treeish objects for ports.\n%s", maybe_output.error());
+        return std::move(maybe_output)
```
Comment (ras0219-msft) [inline src/vcpkg/vcpkgpaths.cpp:1107]: This error doesn't include the command line (which is no worse than before).

File: src/vcpkg/vcpkgpaths.cpp [hunk 3]
```diff
@@ -1110,10 +1123,7 @@ namespace vcpkg
         auto maybe_init_output = flatten(cmd_execute_and_capture_output(init_registries_git_dir), Tools::GIT);
         if (!maybe_init_output)
         {
-            return {Strings::format("Error: Failed to initialize local repository %s.\n%s\n",
-                                    work_tree,
-                                    std::move(maybe_init_output).error()),
-                    expected_right_tag};
+            return msg::format_error(msgGitCommandFailed, msg::command_line = init_registries_git_dir.command_line());
```
Comment (ras0219-msft) [inline src/vcpkg/vcpkgpaths.cpp:1126]: This no longer includes the failing command output.

File: src/vcpkg/vcpkgpaths.cpp [hunk 4]
```diff
@@ -1162,9 +1172,9 @@ namespace vcpkg
         auto maybe_init_output = flatten(cmd_execute_and_capture_output(init_registries_git_dir), Tools::GIT);
         if (!maybe_init_output)
         {
-            return Strings::format("Error: Failed to initialize local repository %s.\n%s\n",
-                                   work_tree,
-                                   std::move(maybe_init_output).error());
+            return msg::format_error(msgGitFailedToInitializeLocalRepository, msg::path = work_tree)
```
Comment (ras0219-msft) [inline src/vcpkg/vcpkgpaths.cpp:1175]: Doesn't display the git command line (no regression).

File: src/vcpkg/vcpkgpaths.cpp [hunk 5]
```diff
@@ -1177,10 +1187,9 @@ namespace vcpkg
         auto maybe_fetch_output = flatten(cmd_execute_and_capture_output(fetch_git_ref), Tools::GIT);
         if (!maybe_fetch_output)
         {
-            return Strings::format("Error: Failed to fetch ref %s from repository %s.\n%s\n",
-                                   treeish,
-                                   repo,
-                                   std::move(maybe_fetch_output).error());
+            return msg::format_error(msgGitFailedToFetch, msg::value = treeish, msg::url = repo)
```
Comment (ras0219-msft) [inline src/vcpkg/vcpkgpaths.cpp:1190]: Doesn't display the git command line (no regression)

File: src/vcpkg/vcpkgpaths.cpp [hunk 6]
```diff
@@ -1129,23 +1139,23 @@ namespace vcpkg
         auto maybe_fetch_output = flatten(cmd_execute_and_capture_output(fetch_git_ref), Tools::GIT);
         if (!maybe_fetch_output)
         {
-            return {Strings::format("Error: Failed to fetch ref %s from repository %s.\n%s\n",
-                                    treeish,
-                                    repo,
-                                    std::move(maybe_fetch_output).error()),
-                    expected_right_tag};
+            return msg::format_error(msgGitFailedToFetch, msg::value = treeish, msg::url = repo)
```
Comment (ras0219-msft) [inline src/vcpkg/vcpkgpaths.cpp:1142]: Doesn't display the git command line (no regression)

File: src/vcpkg/vcpkgpaths.cpp [hunk 7]
```diff
@@ -1222,8 +1231,8 @@ namespace vcpkg
 
         auto pid = get_process_id();
 
-        Path git_tree_temp = Strings::format("%s.tmp%ld", git_tree_final, pid);
-        Path git_tree_temp_tar = Strings::format("%s.tmp%ld.tar", git_tree_final, pid);
+        Path git_tree_temp = fmt::format("{}.tmp{}", git_tree_final, pid);
+        Path git_tree_temp_tar = fmt::format("{}.tmp{}.tar", git_tree_final, pid);
```
Comment (ras0219-msft) [inline src/vcpkg/vcpkgpaths.cpp:1235]:
```suggestion
        Path git_tree_temp_tar = fmt::format("{}.tar", git_tree_temp);
```

**Review by ras0219-msft — APPROVED** [review]

File: src/vcpkg/vcpkgpaths.cpp (hunk 2 above)
Comment (BillyONeal) [reply src/vcpkg/vcpkgpaths.cpp:1107]: I put a todo against myself to change the `Command` and `flatten` family to do a better job of uniformly reporting tool name, full command line, exit code, and output. I'll spot fix the ones you pointed out here. (I do note though that substantially changing the output is the exact opposite of what you asked for before :) )

File: src/vcpkg/vcpkgpaths.cpp (hunk 7 above)
Comment (BillyONeal) [reply src/vcpkg/vcpkgpaths.cpp:1235]: Used `operator+` rather than `fmt::format` here.

File: src/vcpkg/vcpkgpaths.cpp (hunk 1 above)
Comment (BillyONeal) [reply src/vcpkg/vcpkgpaths.cpp:1016]: It's now added on 1027.

**Review by ras0219-msft — APPROVED** [review]

**Pull request merged**